From f70123a40193c2056755117aa291582f42db5f35 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
Date: Sun, 13 Apr 2025 17:47:53 +0200
Subject: [PATCH] sqlite3.pc.in: revert upstream commits causing build issues

This reverts upstream commit a2d88f6a03c9ebde3d037b0e297f47216c685381
and a number of following commits related to the same commit. The
sqlite build system is indeed upstream not based on autoconf, but the
project "generates" an "autoconf-based amalgamation":

  https://www3.sqlite.org/src/dir?ci=tip&name=autoconf

Unfortunately commit a2d88f6a03c9ebde3d037b0e297f47216c685381 broke
the build with the autoconf amalgamation because while @LIBS@ gets
replaced by the autoconf machinery, @LDFLAGS_*@ do not, causing build
failures such as:

output-1/host/lib/gcc/sparc-buildroot-linux-uclibc/13.3.0/../../../../sparc-buildroot-linux-uclibc/bin/ld: cannot find @LDFLAGS_MATH@: No such file or directory
output-1/host/lib/gcc/sparc-buildroot-linux-uclibc/13.3.0/../../../../sparc-buildroot-linux-uclibc/bin/ld: cannot find @LDFLAGS_ZLIB@: No such file or directory
output-1/host/lib/gcc/sparc-buildroot-linux-uclibc/13.3.0/../../../../sparc-buildroot-linux-uclibc/bin/ld: cannot find @LDFLAGS_ICU@: No such file or directory

This issue has been reported at
https://sqlite.org/forum/info/e40b9b424a, and upstream says that the
issue has been fixed in the amalgamation script by basically doing the
same as what this commit does:

  https://sqlite.org/src/info/fe47154799bfe

Upstream: https://sqlite.org/forum/info/e40b9b424a
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
---
 sqlite3.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sqlite3.pc.in b/sqlite3.pc.in
index a9f941b1e4..3799671e61 100644
--- a/sqlite3.pc.in
+++ b/sqlite3.pc.in
@@ -9,5 +9,5 @@ Name: SQLite
 Description: SQL database engine
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -lsqlite3
-Libs.private: @LDFLAGS_MATH@ @LDFLAGS_ZLIB@ @LDFLAGS_ICU@
+Libs.private: @LIBS@
 Cflags: -I${includedir}
-- 
2.49.0

